### DISCUSSION TITLE
tailwindcss w-[60px] does not work, w-4 w-1/3 this works again

### DISCUSSION THREAD

**Discussion by eachann1024** [tailwindlabs/tailwindcss#8844] [Help] [answered]
All other commands are fine, only this particular write has problems. I migrated from windicss. I have dereferenced windicss

![img](https://s2.loli.net/2022/07/13/fLgKmUC86sTO7ty.png)

Environment： vue-cli + vue3 + Element PLUS

``` JSON
module.exports = {
  important: true,
  content: ['./public/index.html', './src/**/*.{html,js,ts,jsx,tsx,vue}'],
  theme: {
    extend: {},
  },
  plugins: [],
  corePlugins: {
		preflight: false,
	},
}
```

**Comment by arisac** [comment, marked as answer]
can you pasted the code where `w-[60px]` is applied?

**Reply by eachann1024** [reply]
It doesn't work, and now we found another problem `bg-red-200` command also doesn't work

**Reply by eachann1024** [reply]
I'd like to know what problems this might be causing and what else I can go and troubleshoot

**Reply by arisac** [reply]
can you create a demo repository for reproduction of your issue?
sometime try to re-create an environment to reproduce the problem is a good way to start debugging/troubleshooting.

**Reply by eachann1024** [reply]
The headache is that it installs correctly in other environments. For the demo library, I'll go create a

**Reply by eachann1024** [reply]
[file](https://drive.google.com/file/d/18K30JQsSILex6FJaSNKdxA7sq7JqZDlc/view?usp=sharing)

I just tried it. bg-red-200 can work, but changing the font size is invalid.

**Reply by arisac** [reply]
Your project is kinda messy. But I guess you are using `yarn` to manage packages. 

I'll give you a quick fix to solve your problem, then explain it, and then why your project is messy.

### 1 fix for your problem

Change your `tailwind.config.js`:

```diff
module.exports = {
  important: true,
  darkMode: false, // or 'media' or 'class'
-  content: ['./public/index.html', './src/**/*.{html,js,ts,jsx,tsx,vue}'],
+  mode: 'jit',
+  purge: ['./index.html', './src/**/*.{vue,html,js,ts}'],
  theme: {
    extend: {
      backgroundColor: ['active'],
      borderColor: ['focus-visible', 'first'],
      textColor: ['visited'],
    },
  },
  plugins: [],
  corePlugins: {
		preflight: false,
	},
}
```

- I also removed `jsx, tsx` since your project is `vue`. Not a problem here, but it's better and clear.

In your `App.vue`:

- it's not `font-[23px]`, use `text-[23px]` for arbitrary values for text size


### 2 Explains

- You are using TailwindCSS V2, not V3. So you'll need `mode: 'jit'` for arbitrary values to run in dev mode. V3 does not require that.
- In TailwindCSS V2, it's `purge: []`, not `content: []`
- You need purge `./index.html` in your project. So it monitors the actual changes in `index`. `public/index.html` is static template, won't be monitored for the build.

### 3 Why I'm saying your project is kinda messy

1. You have `package-lock.json` and `yarn.lock` files in your project. Which is not good for manage deps. You need to stick to one pkg manager in your whole dev, test, production evn.
2. Using `npm i` actually made errs in my env. 
3. Since I suspected you are using `yarn`, you are using `@tailwindcss/postcss7-compat` in your deps, which is TailwindCSS V2. But your `package-lock.json` indicates a TailwindCSS V3, which is kinda mess in your deps mgmt.
4. The rest explained in the Fix step, `content` is used in V3, not V2. And you need `jit` mode in V2.
5. I suggest your team to do a little clean up for your repo, deps and configs. And an upgrade to TailwindCSS V3 is good.

**Reply by eachann1024** [reply]
Thank you very much, the problem has been solved. The new project is currently using TailwindCSS V3, the old project is built with vue-cli and uses less, so upgrading to V3 feels like a lot of work!

> This project has been a long time ago. It was previously commissioned by an outsourced team for development. It took a few twists and turns to get to me, and I need to be very careful about upgrading the base layer and doing cleanup.

**Reply by eachann1024** [reply]
@Triggered a new issue, ** recompile to take effect. But I already have jit turned on **
![img](https://s2.loli.net/2022/07/22/VnioAWwHPDfeRIh.png)

```js
module.exports = {
  important: true,
  mode: 'jit',
  purge: ['./public/index.html', './src/**/*.{vue,html,js,ts}'],
  theme: {
    extend: {
    },
  },
  plugins: [],
  corePlugins: {
		// preflight: false,
	},
}


```

@arisac
